MathLib/LinAlg/SparsityPattern.h

```diff
@@ -0,0 +1,23 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATHLIB_LINALG_SPARSITYPATTERN_H
+#define MATHLIB_LINALG_SPARSITYPATTERN_H
+
+#include <vector>
+
+#include "ProcessLib/NumericsConfig.h"
+
+namespace MathLib
+{
+/// A vector telling how many nonzeros there are in each global matrix row.
+using SparsityPattern = std::vector<GlobalIndexType>;
```

Comment:
Only an note s.t. I dont't forget it: For PETSc we could extend the sparsity pattern, because in PETSc matrices there is a pattern for the "diagonal" part and one for the "off-diagonal" part.
http://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Mat/MatCreateAIJ.html